src/shared/firewall-util.c

```diff
@@ -8,7 +8,10 @@
 #include "firewall-util.h"
 #include "firewall-util-private.h"
 
-static enum FirewallBackend FirewallBackendProbe(void) {
+static enum FirewallBackend FirewallBackendProbe(FirewallContext *ctx) {
+        if (fw_nftables_init(ctx) == 0)
+               return FW_BACKEND_NFTABLES;
```

Comment:
should we cache the result?

Reply:
The caller stores the return value in the FirewallContext, the probe function is only called once per context.